test/libsolidity/SyntaxTest.h

```diff
@@ -45,15 +45,21 @@ class SyntaxTest: public AnalysisFramework, public solidity::test::CommonSyntaxT
 	{
 		return std::make_unique<SyntaxTest>(_config.filename, _config.evmVersion, true);
 	}
-	SyntaxTest(std::string const& _filename, langutil::EVMVersion _evmVersion, bool _parserErrorRecovery = false);
+	SyntaxTest(
+		std::string const& _filename,
+		langutil::EVMVersion _evmVersion,
+		bool _parserErrorRecovery = false,
+		langutil::Error::Severity _minSeverity = langutil::Error::Severity::Info
+	);
 
 protected:
 	virtual void setupCompiler();
 	void parseAndAnalyze() override;
 	virtual void filterObtainedErrors();
 
-	bool m_optimiseYul = true;
-	bool m_parserErrorRecovery = false;
+	bool m_optimiseYul{};
```

Comment:
I see that `m_optimiseYul` defaults to `true` in the constructor, so seems harmless, but still shouldn't we keep it initialized the same value here too?

Reply:
It does not really default to `true`. It defaults to `m_reader.boolSetting("optimize-yul", true)`, which could end up being either. `true` is just as misleading here as `false`.

I think that using default initialization here is the least bad option. It does not show the value so it does not draw attention away from the constructor. The actual value here is irrelevant, we only really care that it's initialized to some deterministic value.